### ISSUE TITLE
Add comprehensive documentation

### ISSUE THREAD

**Issue by ryanjulian** [rlworkgroup/garage#1426]
This is an umbrella issue, as this will be a large project.

Overall outline format inspired by https://documentation.divio.com/

* Installation
* Getting Started
* Usage Guide (How-To)
  - Train a policy to solve and environment
  - Use image observations
  - Load and use a trained policy
  - Use a pre-trained network to start a new experiment
  - Monitor your experiments with TensorBoard
  - Save, load, and resume experiments
  - Maximize resource usage (CPUs, GPUs, etc.)
  - Run many experiments
  - Distribute experiments across machines
  - Run Garage with Docker
  - Ensure your experiment are reproducible
  - Run meta-/multi-task RL experiments
* Tutorials
  - Add a new environment
  - Add a new algorithm —  PyTorch / TensorFlow / numpy
  - Implement a custom Policy — PyTorch / TensorFlow / numpy
  - Implement a custom Q-Function — PyTorch / TensorFlow / numpy
  - Implement a custom Value Function — PyTorch / TensorFlow / numpy
  - Implement a custom Encoder/Decoder — PyTorch / TensorFlow / numpy
  - Change how your algorithm samples (Implement a Custom Worker)
* Algorithms
  - Blackbox (gradient-free)
    * Cross-Entropy Method (CEM) — numpy
    * CMA-ES — numpy
    * Relative Entropy Policy Search (REPS) — TensorFlow
  - On-Policy
    * Proximal Policy Optimization (PPO) — PyTorch, TensorFlow
    * Trust Region Policy Optimization (TRPO) — PyTorch, TensorFlow
    * REINFORCE — PyTorch, TensorFlow
    * Episodic Reward Weighted Regression (ERWR) — TensorFlow
    * Truncated Natural Policy Gradient (TNPG)
  - Off-Policy Discrete Action-Space
    * (Double-)Deep Q-Networks/Double-Deep Q-Networks (DQN/DDQN) — TensorFlow
  - Off-Policy Continuous Action-Space
    * Deep Deterministic Policy Gradients (DDPG) — PyTorch, TensorFlow
    * Twin Delayed DDPG (TD3) — TensorFlow
    * Soft Actor-Critic (SAC) — PyTorch
  - Multi-Task RL
    * Multi-Task PPO MTPPO — PyTorch, TensorFlow
    * Multi-Task TRPO (MTTRPO) — PyTorch, TensorFlow
    * Multi-Task SAC (MTSAC) — PyTorch
    * Task Embeddings with PPO (TEPPO) — TensorFlow
  - Meta-RL
    * Fast Reinforcement Learning via Slow Reinforcement Learning (RL^2) — TensorFlow
    * Probabilistic Embeddings for Actor-Critic RL (PEARL) — PyTorch
    * Model-Agnostic Meta-Learning (MAML) — PyTorch
  - Imitation Learning
    * Behavioral cloning — numpy
* Tips and Tricks
  - NaNs or Infs in your algorithm
* RL Resources
* Reference Guide
  - Core Concepts
    * Experiment
    * Environment
    * Models
      - Policy
      - ValueFunction
      - QFunction
      - Encoder/Decoder
    * Optimizer
    * ReplayBuffer
    * Sampling
      - Sampler
      - Worker
    * Data Structures
      - TrajectoryBatch
      - Sample and SampleBatch
    * Trainer (nee LocalRunner)
    * Evaluator
    * Snapshotter
  - Evaluation
  - Logging and Plotting
  - Framework-specific notes
    * numpy-specifics
    * TensorFlow
    * PyTorch
* Developing Garage
  - Setting up your development environment
  - Development workflow
  - Testing
  - Benchmarking
  - Writing Documentation
  - Preparing a Pull Request

**Comment by ryanjulian** [comment]
Status as of 2020.09rc1

This is an umbrella issue, as this will be a large project.

Overall outline format inspired by https://documentation.divio.com/

* [ ] Installation
* [ ] Getting Started
* [ ] Usage Guide (How-To)
  - [ ] Train a policy to solve an environment
  - [x] Use image observations
  - [ ] Load and use a trained policy
  - [ ] Use a pre-trained network to start a new experiment
  - [ ] Monitor your experiments with TensorBoard
  - [ ] Save, load, and resume experiments
  - [ ] Maximize resource usage (CPUs, GPUs, etc.)
  - [ ] Run many experiments
  - [ ] _Distribute experiments across machines_
  - [ ] Run Garage with Docker
  - [ ] Ensure your experiment are reproducible
  - [ ] Run meta-/multi-task RL experiments
* [ ] Tutorials
  - [ ] Add a new environment
  - [ ] Add a new algorithm —  PyTorch / TensorFlow / numpy
  - [ ] Implement a custom Policy — PyTorch / TensorFlow / numpy
  - [ ] Implement a custom Q-Function — PyTorch / TensorFlow / numpy
  - [ ] Implement a custom Value Function — PyTorch / TensorFlow / numpy
  - [ ] Implement a custom Encoder/Decoder — PyTorch / TensorFlow / numpy
  - [ ] Change how your algorithm samples (Implement a Custom Worker)
  - [ ] Using a publicly-available pre-trained network (ResNet, etc.) as part of a Model
* [ ] Environment Libraries
  - [ ] OpenAI Gym
  - [ ] dm_control
  - [ ] MetaWorld
  - [ ] PyBullet
* [ ] Algorithms
  - [ ] Blackbox (gradient-free)
    * [ ] Cross-Entropy Method (CEM) — numpy
    * [ ] CMA-ES — numpy
    * [ ] Relative Entropy Policy Search (REPS) — TensorFlow
  - [ ] On-Policy
    * [ ] Proximal Policy Optimization (PPO) — PyTorch, TensorFlow
    * [ ] Trust Region Policy Optimization (TRPO) — PyTorch, TensorFlow
    * [ ] REINFORCE — PyTorch, TensorFlow
    * [ ] Episodic Reward Weighted Regression (ERWR) — TensorFlow
    * [ ] Truncated Natural Policy Gradient (TNPG)
  - [ ] Off-Policy Discrete Action-Space
    * [ ] (Double-)Deep Q-Networks/Double-Deep Q-Networks (DQN/DDQN) — TensorFlow
  - [ ] Off-Policy Continuous Action-Space
    * [ ] Deep Deterministic Policy Gradients (DDPG) — PyTorch, TensorFlow
    * [ ] Twin Delayed DDPG (TD3) — TensorFlow
    * [ ] Soft Actor-Critic (SAC) — PyTorch
  - [ ] Multi-Task RL
    * [ ] Multi-Task PPO MTPPO — PyTorch, TensorFlow
    * [ ] Multi-Task TRPO (MTTRPO) — PyTorch, TensorFlow
    * [ ] Multi-Task SAC (MTSAC) — PyTorch
    * [ ] Task Embeddings with PPO (TEPPO) — TensorFlow
  - [ ] Meta-RL
    * [ ] Fast Reinforcement Learning via Slow Reinforcement Learning (RL^2) — TensorFlow
    * [ ] Probabilistic Embeddings for Actor-Critic RL (PEARL) — PyTorch
    * [ ] Model-Agnostic Meta-Learning (MAML) — PyTorch
  - [ ] Imitation Learning
    * [ ] Behavioral cloning — numpy
* [ ] Tips and Tricks
  - [ ] NaNs or Infs in your algorithm
* [ ] RL Resources
* [ ] Reference Guide
  - [ ] Core Concepts
    * [ ] Experiment
    * [ ] Environment
    * [ ] Models
      - [ ] Policy
      - [ ] ValueFunction
      - [ ] QFunction
      - [ ] Encoder/Decoder
    * [ ] Optimizer
    * [ ] ReplayBuffer
    * [ ] Sampling
      - [ ] Sampler
      - [ ] Worker
    * [ ] Data Structures
      - [ ] TrajectoryBatch
      - [ ] Sample and SampleBatch
    * [ ] Trainer (nee LocalRunner)
    * [ ] Evaluator
    * [ ] Snapshotter
  - [ ] Evaluation
  - [ ] Logging and Plotting
  - [ ] Framework-specific notes
    * [ ] numpy-specifics
    * [ ] TensorFlow
    * [ ] PyTorch
* [ ] Developing Garage
  - [ ] Setting up your development environment
  - [ ] Development workflow
  - [x] Testing
  - [x] Benchmarking
  - [ ] Writing Documentation
  - [ ] Preparing a Pull Request
* [ ] API Reference (autogenerated)

**Comment by ryanjulian** [comment]
Pages for 2020.09rc2:
* Usage Guide > Run many experiments
* Development Guide > Development workflow
* Development Guide > Preparing a Pull Request
* Reference Guide > Data Structures
* Usage Guide > Run meta/multi-task RL experiments
* Usage Guide > Maximize resource usage
* Getting Started
* Usage Guide > Use a pre-trained network to start a new experiment
* Usage Guide > Run garage with Docker
* Tutorials > Using a publicly-available pre-trained network (ResNet, etc.) as part of a Model
* Tutorials > Change how your algorithm samples (Implement a custom worker)
* Tutorials > Implement a custom Algorithm (TF and Numpy sections)
* Tutorials > Adding a new Environment (updated for garage.Env)
* Environment Libraries > {OpenAI Gym, dm_control, MetaWorld, PyBullet}
* Algorithms > Behavioral Cloning -- PyTorch
* Algorithms > PCGrad -- {TensorFlow, PyTorch}
* Algorithms > TD3 -- {TensorFlow, PyTorch}
* Algorithms > A2C -- {TensorFlow, PyTorch}
* Algorithms > A3C -- {PyTorch}
